shuffle.py

```diff
@@ -34,7 +32,7 @@ def get_shuffle_params(params, index):
     random.seed(0)
     keys = sorted(params.keys())
     iters = [iter_shuffle(params[key]) for key in keys]
-    vals = next(itertools.islice(six.moves.zip(*iters), index, None))
+    vals = tuple(iter[index] for iter in iters)
```

Comment:
I'm sorry.  Iterators don't support getitem.
```suggestion
    vals = tuple(next(itertools.islice(iter, index, None)) for iter in iters)
```